app/main.py

```diff
@@ -1,2 +1,57 @@
+from typing import Hashable
+
+
+class Node:
+    def __init__(self, key: Hashable, value: any) -> None:
+        self.key = key
+        self.value = value
+        self.hash = hash(key)
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.load_factor = 2 / 3
+        self.length = 0
+        self.hash_table: list[None | list[Node]] = [None] * self.capacity
+
+    def __setitem__(self, key: Hashable, value: any) -> None:
+        if self.length / self.capacity >= self.load_factor:
+            self.resize()
+
+        idx = hash(key) % self.capacity
+
+        if self.hash_table[idx] is None:
+            self.hash_table[idx] = []
+
+        for node in self.hash_table[idx]:
+            if node.key == key:
+                node.value = value
+                return
+
+        self.hash_table[idx].append(Node(key, value))
+        self.length += 1
+
+    def __getitem__(self, key: Hashable) -> None:
+        idx = hash(key) % self.capacity
+
+        if self.hash_table[idx]:
+            for node in self.hash_table[idx]:
+                if node.key == key:
+                    return node.value
+        raise KeyError
```

Comment:
When raising a KeyError, it's a good practice to include the key that caused the error, e.g., raise KeyError(key).